### PR TITLE
BDRE-312 	  New settings page

### PR DESCRIPTION
**Pull request by GithubMrinmoy**
BDRE-312 New settings page

### CODE REVIEW

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java [hunk 1]
```diff
@@ -175,6 +275,27 @@ RestWrapper delete(@PathVariable("cg") String configGroup, @PathVariable("key")
...
 
+    /**
+     * This method fetches all distinct record in GeneralConfig table depending on Config group.
+     *
+     * @return restWrapper with the list of instance of GeneralConfig table.
+     */
+
+    @RequestMapping( method = RequestMethod.GET)
+    @ResponseBody public
+    RestWrapper getDistinctGenerelConfig(Principal principal) {
```
Comment (bdre-bda) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java]: MAJOR Remove the unused method parameter(s) "principal". [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1172)

**Comment by bdre-bda** [comment]
SonarQube analysis reported 10 issues:
* MAJOR 1 major
* MINOR 9 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L158) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L197) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L136) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Define a constant instead of duplicating this literal "enabled" 4 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L131) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L219) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00105)
* MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00105)
* MINOR [Define a constant instead of duplicating this literal " and config group:" 3 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java#L138) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal " Record with key:" 3 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/2201403d2bff8fb93e0e89b7e3db5f6c37416af5/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java#L138) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java (hunk 1 above)
Comment (GithubMrinmoy) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java:286]: @mishisinku Should I remove Principal from method signature?

**Comment by GithubMrinmoy** [comment]
@bdrekapil @mishisinku Please review and merge.

File: md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java (hunk 1 above)
Comment (mishisinku) [inline md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java:286]: @GithubMrinmoy  Please use principal in LOGGER statement. Please refer to any other restAPI already written.

**Comment by bdre-bda** [comment]
SonarQube analysis reported 9 issues:
- MINOR 9 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
- MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L158) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
- MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00105)
- MINOR [Define a constant instead of duplicating this literal "enabled" 4 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L131) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
- MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L136) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
- MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L197) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
- MINOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java#L219) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)
- MINOR [Define a constant instead of duplicating this literal " Record with key:" 3 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java#L138) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
- MINOR [Define a constant instead of duplicating this literal " and config group:" 3 times.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java#L138) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
- MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/WiproOpenSourcePractice/openbdre/blob/b0f30022adbfea2f956f35253e65cb4967c3560b/md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/GeneralConfigAPI.java) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00105)

**Pull request merged**